packages/strapi-hook-mongoose/lib/index.js

```diff
@@ -422,7 +430,7 @@ module.exports = function (strapi) {
                     kind: String,
                     [details.filter]: String,
                     ref: {
-                      type: instance.Schema.Types.ObjectId,
+                      type: String,
```

Comment:
Why is this hardcoded to `String`? I suppose it should be using the `keyType` variable?

Reply:
I'had a case where some entities have `String` as key, and others have `ID`, so this special fixed `String` is for having that case, and queries on the associations should just work (tm)